sparse/unit_test/Test_Sparse_spgemm.hpp

```diff
@@ -113,7 +114,14 @@ int run_spgemm(crsMat_t A, crsMat_t B,
     EXPECT_TRUE(sh->is_numeric_called());
 
     if (testReuse) {
-      // Give A and B completely new random values, and re-run just numeric
+      // Give A and B completely new random values (changing both the pointer
```

Comment:
Ideally there should be a test that catches the throw from the spgemm handle to make sure that when the handle needs to be rebuilt it does the right thing.

Reply:
@lucbv I added a test that exercises the new checks